src/Microsoft.AspNet.Razor/Parser/SyntaxTree/BlockBuilder.cs

```diff
@@ -21,7 +21,7 @@ public BlockBuilder(Block original)
         }
 
         public BlockType? Type { get; set; }
-        public IList<SyntaxTreeNode> Children { get; private set; }
+        public List<SyntaxTreeNode> Children { get; private set; }
```

Comment:
But `Block` doesn't need to change the collection does it? `IReadOnlyList` should work here w/ a `List` backing field.